Library/Homebrew/cmd/update.sh

```diff
@@ -331,6 +345,13 @@ EOS
     UPSTREAM_BRANCH="$(upstream_branch)"
     # the refspec ensures that the default upstream branch gets updated
     (
+      if [[ -n "$HOMEBREW_UPDATE_PREINSTALL" ]]
+      then
+        # Skip taps without formulae.
+        FORMULAE="$(find "$DIR" -maxdepth 1 -name '*.rb' -or -name 'Formula' -print -quit)"
```

Comment:
You'd also need to check for a `HomebrewFormula` directory for consistency with `Tap#formula_dir` and possibly even for a `Casks` directory if we wanted this to also impact the `brew install caskroom/<repo>/<cask>` case. Not sure about the latter though.